File: lista Go/at2.go
package main

import "fmt"


type Carro struct {
    Marca     string
    Modelo    string
    Ano       int
    Velocidade int
}


func (c *Carro) Acelerar(incremento int) {
    c.Velocidade += incremento
    fmt.Printf("%s acelerou. Velocidade atual: %d km/h\n", c.Modelo, c.Velocidade)
}


func (c *Carro) Frear(decremento int) {
    c.Velocidade -= decremento
    if c.Velocidade < 0 {
        c.Velocidade = 0  
    }
    fmt.Printf("%s freou. Velocidade atual: %d km/h\n", c.Modelo, c.Velocidade)
}


func (c Carro) ExibirVelocidade() {
    fmt.Printf("Velocidade atual de %s: %d km/h\n", c.Modelo, c.Velocidade)
}

func main() {
    
    carro1 := Carro{Marca: "Toyota", Modelo: "Corolla", Ano: 2020, Velocidade: 0}

   
    carro1.Acelerar(30)
    carro1.Acelerar(50)
    carro1.Frear(20)
    carro1.ExibirVelocidade()
}

File: lista java/at2.java
public class Carro {
    
    private String marca;
    private String modelo;
    private int ano;
    private int velocidade;

    
    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;  
    }

  
    public void acelerar(int incremento) {
        this.velocidade += incremento;
        System.out.println(modelo + " acelerou. Velocidade atual: " + velocidade + " km/h");
    }

    
    public void frear(int decremento) {
        this.velocidade -= decremento;
        if (velocidade < 0) {
            velocidade = 0;  
        }
        System.out.println(modelo + " freou. Velocidade atual: " + velocidade + " km/h");
    }

    
    public void exibirVelocidade() {
        System.out.println("Velocidade atual de " + modelo + ": " + velocidade + " km/h");
    }

    public static void main(String[] args) {
       
        Carro carro1 = new Carro("Toyota", "Corolla", 2020);

        
        carro1.acelerar(30);
        carro1.acelerar(50);
        carro1.frear(20);
        carro1.exibirVelocidade();
    }
}
